File: challenges/easy/06_date_format/buggy.py
import datetime


def days_between(start_str, end_str):
    """
    Given two date strings in DD/MM/YYYY format,
    return the number of days from start to end.
    """
    fmt = "%m/%d/%Y"
    start = datetime.datetime.strptime(start_str, fmt)
    end = datetime.datetime.strptime(end_str, fmt)
    return (end - start).days

File: challenges/easy/20_date_format_ts/buggy.ts
export function daysBetween(startStr: string, endStr: string): number {
  const parseDate = (str: string): Date => {
    const parts = str.split('/');
    return new Date(Number(parts[2]), Number(parts[0]) - 1, Number(parts[1]));
  };
  const start = parseDate(startStr);
  const end = parseDate(endStr);
  return Math.round((end.getTime() - start.getTime()) / (1000 * 60 * 60 * 24));
}
